geomesa-hbase/geomesa-hbase-datastore/src/main/java/org/locationtech/geomesa/hbase/filters/JSimpleFeatureFilter.java

```diff
@@ -65,22 +80,31 @@ public ReturnCode filterKeyValue(Cell v) throws IOException {
     }
 
     private static class CQLFilter extends AbstractFilter {
-        KryoBufferSimpleFeature reusable;
-        CQLFilter(String sftString, String filter) {
-            super(sftString, filter);
-            reusable = IteratorCache.serializer(sftString, SerializationOptions.withoutId()).getReusableFeature();
+        CQLFilter(String sftString, String filter, String transform, String transformSchema) {
+            super(sftString, filter, transform, transformSchema);
         }
 
         @Override
         public ReturnCode filterKeyValue(Cell v) throws IOException {
             try {
                 log.trace("cell = {}", v);
-                reusable.setBuffer(v.getValueArray(), v.getValueOffset(), v.getValueLength());
-                log.trace("Evaluating filter against SimpleFeature");
-                if (filter.evaluate(reusable)) {
-                    return ReturnCode.INCLUDE;
+                if (hasTransform) {
+                    reusable.setBuffer(CellUtil.cloneValue(v));
+                    log.trace("Evaluating filter against SimpleFeature");
+                    if (filter.evaluate(reusable)) {
+                        return ReturnCode.INCLUDE;
+                    } else {
+                        return ReturnCode.SKIP;
+                    }
                 } else {
-                    return ReturnCode.SKIP;
+                    KryoFeatureSerializer serializer = new KryoFeatureSerializer(sft, SerializationOptions.withoutId());
+                    SimpleFeature sf = serializer.deserialize(v.getValueArray(), v.getValueOffset(), v.getValueLength());
```

Comment:
These two lines can likely be re-written as the original
reusable.setBuffer(v.getValueArray(), v.getValueOffset(), v.getValueLength());